File: SeekMatch-Client/src/app/shared/models/talent.ts
import { Education } from "./education";
import { Experience } from "./experience";
import { Setting } from "./setting";
import { User } from "./user";

export class Talent extends User {

  public firstName: string = '';
  public lastName: string = '';
  public password: string= '';
  public profileTitle : string= '';
  public summary : string= '';
  public dateOfBirth : string | null = null;
  public phone : string= '';
  public country : string= '';
  public region : number | null = null;
  public city : number | null = null;
  public website : string = '';
  public linkedIn : string = '';
  public profilePicture : Uint8Array | null = null;
  public educations : Education[] = [];
  public experiences : Experience[] = [];
  public setting: Setting = new Setting();

  constructor(init?:Partial<Talent>) {
    super();
    Object.assign(this, init);
  }
}

File: SeekMatch-Server/SeekMatch.Core/Entities/Talent/Talent.cs
namespace SeekMatch.Core.Entities
{
    public class Talent: BaseEntity
    {
        public required string FirstName { get; set; }
        public required string LastName { get; set; }
        public string? ProfileTitle { get; set; }
        public string? Summary { get; set; }
        public DateOnly DateOfBirth { get; set; }
        public string? Phone { get; set; }
        public string? Country { get; set; }
        public int? Region { get; set; }
        public int? City { get; set; }
        public string? Website { get; set; }
        public string? LinkedIn { get; set; }
        public byte[]? ProfilePicture { get; set; }
        public required User User { get; set; }
        public List<Education> Educations { get; set; } = new List<Education>();
        public List<Experience> Experiences { get; set; } = new List<Experience>();
        public List<JobApplication> JobApplications { get; set; } = new List<JobApplication>();
    }
}
